pkg/scalers/couchdb_scaler.go

```diff
@@ -0,0 +1,247 @@
+package scalers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	couchdb "github.com/go-kivik/couchdb/v3"
+	"github.com/go-kivik/kivik/v3"
+	"github.com/go-logr/logr"
+	kedautil "github.com/kedacore/keda/v2/pkg/util"
+	v2 "k8s.io/api/autoscaling/v2"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/metrics/pkg/apis/external_metrics"
+	"strconv"
+)
+
+type couchDBScaler struct {
+	metricType v2.MetricTargetType
+	metadata   *couchDBMetadata
+	client     *kivik.Client
+	logger     logr.Logger
+}
+
+type couchDBQueryRequest struct {
+	Selector map[string]interface{} `json:"selector"`
+	Fields   []string               `json:"fields"`
+}
+
+type couchDBMetadata struct {
+	connectionString     string
+	host                 string
+	port                 string
+	username             string
+	password             string
+	dbName               string
+	query                string
+	queryValue           int64
+	activationQueryValue int64
+	metricName           string
+	scalerIndex          int
+}
+
+type Res struct {
+	ID       string `json:"_id"`
+	Feet     int    `json:"feet"`
+	Greeting string `json:"greeting"`
+}
+
+func (s *couchDBScaler) GetMetricSpecForScaling(context.Context) []v2.MetricSpec {
+	externalMetric := &v2.ExternalMetricSource{
+		Metric: v2.MetricIdentifier{
+			Name: GenerateMetricNameWithIndex(s.metadata.scalerIndex, s.metadata.metricName),
+		},
+		Target: GetMetricTarget(s.metricType, s.metadata.queryValue),
+	}
+	metricSpec := v2.MetricSpec{
+		External: externalMetric, Type: externalMetricType,
+	}
+	return []v2.MetricSpec{metricSpec}
+}
+
+func (s couchDBScaler) Close(ctx context.Context) error {
+	if s.client != nil {
+		err := s.client.Close(ctx)
+		if err != nil {
+			s.logger.Error(err, fmt.Sprintf("failed to close couchdb connection, because of %v", err))
+			return err
+		}
+	}
+	return nil
+}
+
+func (s *couchDBScaler) getQueryResult(ctx context.Context) (int64, error) {
+	db := s.client.DB(ctx, s.metadata.dbName)
+	var request couchDBQueryRequest
+	err := json.Unmarshal([]byte(s.metadata.query), &request)
+	if err != nil {
+		fmt.Println("Couldn't unmarshal query string: ", err)
+		s.logger.Error(err, fmt.Sprintf("Couldn't unmarshal query string because of %v", err))
+		return 0, err
+	}
+	rows, err := db.Find(ctx, request, nil)
+	if err != nil {
+		fmt.Println("Couldn't fetch rows: ", err)
+		s.logger.Error(err, fmt.Sprintf("failed to fetch rows because of %v", err))
+		return 0, err
+	}
+	var count int64
+	for rows.Next() {
+		count++
+		res := Res{}
+		if err := rows.ScanDoc(&res); err != nil {
+			s.logger.Error(err, fmt.Sprintf("failed to scan the doc because of %v", err))
+			fmt.Println("Error while scanning the doc, err: ", err)
+			return 0, err
+		}
+		fmt.Println(res)
+	}
```

Comment:
Should we enforce a scalar as returning value instead of iterating and counting the rows? For multiple rows count, it's possible to use [`_count`](https://docs.couchdb.org/en/stable/maintenance/performance.html#built-in-reduce-functions) instead of iterating all the rows, and if we set the constraint of returning a scalar value, end-users can use the query to get the count or the value of a specific row.